src/Services/Thumbnailer.vala

```diff
@@ -88,30 +82,12 @@ public class PaperClip.Services.Thumbnailer {
         float scaled_height = height * scale_factor, scaled_width = width * scale_factor;
         debug (@"Height: $height -> $scaled_height. Width: $width -> $scaled_width");
 
-        var pixbuf = new Gdk.Pixbuf.from_file_at_size (save_path, (int) scaled_width, (int) scaled_height);
-        var texture = Gdk.Texture.for_pixbuf (pixbuf);
+        var pixbuf = Gdk.pixbuf_get_from_surface (surface, 0, 0, width, height);
+        var scaled_pixbuf = pixbuf.scale_simple ((int) scaled_width, (int) scaled_height, Gdk.InterpType.BILINEAR);
```

Comment:
Unfortunately, GdkPixbuf's scaling provides a lower quality thumbnail. In the DocumentThumbnail widget we use either a Trilinear interpolation or a nearest interpolation. The latter is supported by GdkPixbuf but the former is not.

I also want to remove GdkPixbuf from the codebase in favor of [Glycin](https://gnome.pages.gitlab.gnome.org/glycin/libglycin-gtk4/) since it plays better with GdkTextures and would allow us to remove this ugly scaling implementation I made.

 https://github.com/Diego-Ivan/Paper-Clip/blob/7777db79e327fc7738af1ccd6a49df5374d2cfcd/src/DocumentThumbnail.vala#L73-L78